### ISSUE TITLE
Monthly Milestone - Feb 2018

### ISSUE THREAD

**Issue by kjdelisle** [strongloop/loopback-next#927]
**Tech debt**
  - [x] 🏃 Drop Node 6 support #611
    - blocked by https://github.com/strongloop/loopback-next/issues/959
  - [x] Bring examples back into the monorepo #836 
    - blog post WIP https://github.com/strongloop/strongloop.com/pull/110

**OpenAPI**
- [ ] 🏃 Switch to OpenAPI v3 for spec and routing - #753
    - needs to add tests - #181 

**`@loopback/boot`**

- [ ] 🏃 Automatic configuration of controllers in Application - #780
- [x] 🏃 Automatic configuration of repositories in Application - #781

**Developer Experience**
- [x] 🏃 Circular dependencies in bindings #535 

💪💪 Stretch goals 💪💪
- Update the documentation to align with the current state of the alpha
  - Fix outdated documentation
  - Mark items whose support is outside MVP so that users know these features are not yet live (rather than removing and rewriting those docs later) #796
- Ramp up community participation
  - Create/update policy regarding community contributors and commit rights
  - Outline the new process we intend to follow for planning work and meeting deliverables
  - Set some minimum requirements for community contributors to earn commit rights https://github.com/strongloop/loopback-next/issues/943
  - Gather feedback for subsequent milestones (future planning)

**CLI**
- [x] [CLI] Remove copyright from generated template instances - #944 

**Docs**
- [x] Move contribution doc from wiki to repo #574

**Comment by kjdelisle** [comment]
I think we should add "Drop Node 6 support" to this month's goals, since the longer we put it off, the more the scope of the work is increasing.
(side note: I also think we should abandon the "Monthly Milestone" label, and actually use the milestone feature to track this stuff instead)

**Comment by virkt25** [comment]
I'd also suggest adding this since it's CI related: https://github.com/strongloop/loopback-next/issues/879

**Comment by kjdelisle** [comment]
Updated with suggestions (keeping in mind that they may still be removed)

**Comment by dhmlau** [comment]
A few questions/comments: (maybe you're planning to flush out the details tomorrow, but let me put it here just in case)

> Fix outdated documentation

How are we going to determine the list of items?  everything in the Docs epic?  I just found out that the FAQ page is very out-of-date.   

> Mark items whose support is outside MVP so that users know these features are not yet live (rather than removing and rewriting those docs later)

Not sure if I misunderstood you.. but don't we have `MVP` tag for all the MVP stuff and supposedly everything else is non-MVP? 

> Set some minimum requirements for community contributors to earn commit rights

At some point, we (@raymondfeng , @bajtos , @kjdelisle and others) had talked about the requirements to be a maintainer, if that's what you're referring to.  The notes is in box, can share it here or whatever github issue created. 

Also, i guess the goal for Feb milestone include getting all items (whichever still fits in MVP scope) out of the way too?

**Comment by kjdelisle** [comment]
>Not sure if I misunderstood you.. but don't we have MVP tag for all the MVP stuff and supposedly everything else is non-MVP?

As an example, the current top-down story is not yet supported and won't be for MVP. But we've still got docs to that effect that need to be marked with a warning/notice/header of some kind that tells new users that this isn't yet ready.

**Comment by kjdelisle** [comment]
>Also, i guess the goal for Feb milestone include getting all items (whichever still fits in MVP scope) out of the way too?

There is no way that we can finish all of those items by end of Feb. We've got 275 story points in the MVP release, and that also includes items without estimates: https://github.com/strongloop/loopback-next/issues/927#reports?report=release&release=59fb46b9d83fbd6b606efc3d

**Comment by dhmlau** [comment]
> Also, i guess the goal for Feb milestone include getting all items (whichever still fits in MVP scope) out of the way too?
sorry..  what i mean is to have all the outstanding Jan milestone items done, as one of the goals for the Feb milestone.

**Comment by virkt25** [comment]
Here's another proposal: https://github.com/strongloop/loopback-next/issues/585

**Comment by bajtos** [comment]
Thank you @kjdelisle for kicking off the conversation about the priorities for our February milestone. Your proposal looks good to me 👍 

(side note: I also think we should abandon the "Monthly Milestone" label, and actually use the milestone feature to track tasks related to the goals instead)

Please, don't leave side notes that can derail the conversation, open a new issue instead! That way this issue can stay focused on the monthly milestone and the conversation about the mechanics of using GitHub to track monthly milestones can stay in one place. 

I created the issue myself, see https://github.com/strongloop/loopback-next/issues/930

**Comment by kjdelisle** [comment]
>sorry.. what i mean is to have all the outstanding Jan milestone items done, as one of the goals for the Feb milestone.

@dhmlau 
We need to follow a process like this, IMO:
- Tentatively add items to the milestone (community suggestions welcome!)
- Gather up all the issues that represent work for these items (or create them if they don't exist)
- Estimate those issues and compare to our current velocity
- Rescope the milestone based on what we *can* accomplish vs. what we should be trying to accomplish
- Move the extra items to stretch goals (unless there's too many, we'll deal with that as it happens)

**Comment by dhmlau** [comment]
removed ` TypeScript to JSON Schema - #786` for Feb milestone, as @shimks just finished it!

**Comment by jannyHou** [comment]
@kjdelisle And v3 upgrading https://github.com/strongloop/loopback-next/issues/753 in progress, thanks.

**Comment by kjdelisle** [comment]
We've reached the end of Feb.
